common/src/main/java/org/opencds/cqf/ruler/common/utility/ResourceCreator.java

```diff
@@ -0,0 +1,29 @@
+package org.opencds.cqf.ruler.common.utility;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.instance.model.api.IIdType;
+
+public interface ResourceCreator extends FhirContextUser {
+	@SuppressWarnings("unchecked")
+	default <T extends IBaseResource, I extends IIdType> T newResource(I theId) {
```

Comment:
This is a helper that lets you create a new resource of the correct type and version based only on the id.